File: summary/ruby/8_least_common_multiple.rb
def gcd(a, b)
  while a >= 1 && b >= 1
    if a >= b
      a = a % b
    else
      b = b % a
    end
  end

  a >= 1 ? a : b
end

def lcd(a, b)
  (a * b) / gcd(a, b)
end

a, b = gets.chomp.split.map(&:to_i)
puts "#{lcd(a, b)}"

File: summary/go/8_least_common_multiple.go
package main

import "fmt"

func gcd(a, b int) int {
	for a >= 1 && b >= 1 {
		if a >= b {
			a = a % b
		} else {
			b = b % a
		}
	}
	if a >= 1 {
		return a
	} else {
		return b
	}
}

func lcd(a, b int) int {
	return (a * b) / gcd(a, b)
}

func main() {
	var a, b int
	fmt.Scanf("%d %d", &a, &b)
	fmt.Printf("%d\n", lcd(a, b))
}
